modules/living_spaces_event/src/Controller/LivingSpacesEventIcalController.php

```diff
@@ -0,0 +1,151 @@
+    $this->fileRepository = $file_repository;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('database'),
+      $container->get('request_stack'),
+      $container->get('file_system'),
+      $container->get('file.repository')
+    );
+  }
+
+  /**
+   * Callback for 'iCal' route.
+   */
+  public function download(GroupInterface $group) {
+    $host = $this->request->getCurrentRequest()->getHost();
+    $calendar = new Calendar($host);
+
+    $query = $this->database->select('living_spaces_event_field_data', 'efd');
+    $query->fields('efd', ['id', 'label', 'description__value', 'location__value']);
+    $query->condition('efd.space', $group->id());
+    $query->innerJoin('living_spaces_event__field_start_date', 'efsd', 'efsd.entity_id = efd.id');
+    $query->addField('efsd', 'field_start_date_value');
+    $query->innerJoin('living_spaces_event__field_end_date', 'efed', 'efed.entity_id = efd.id');
+    $query->addField('efed', 'field_end_date_value');
+
+    if ($results = $query->execute()->fetchAllAssoc('id')) {
+      foreach ($results as $result) {
+        $event = new Event();
+        $event->setDtStart(new \DateTime($result->field_start_date_value, new \DateTimeZone('UTC')));
+        $event->setDtEnd(new \DateTime($result->field_end_date_value, new \DateTimeZone('UTC')));
+        $event->setSummary($result->label);
+        $event->setDescription($result->description__value);
+        $event->setLocation($result->location__value);
+        $event->setUrl( "{$host}/living-spaces-event/{$result->id}");
+        $calendar->addComponent($event);
+      }
+    }
+
+    $filename = "iCal {$group->label()}.ics";
+    $uri = 'public://' . $filename;
+
+    $content = $calendar->render();
+    try {
+      $file = $this->fileRepository->writeData($content, $uri, FileSystemInterface::EXISTS_REPLACE);
+    }
+    catch (\Exception $e) {
+      throw new NotFoundHttpException();
+    }
+
+    $parts = explode('://', $uri);
+    $filepath = $this->fileSystem->realpath('public://') . '/' . $parts[1];
+
+    if (!file_exists($filepath)) {
+      throw new NotFoundHttpException();
+    }
+
+    $headers = [
+      'Content-Type' => 'text/calendar',
+      'Content-Disposition' => 'attachment; filename="' . $file->getFilename() . '"',
+      'Content-Length' => $file->getSize(),
+      'Content-Transfer-Encoding' => 'binary',
+      'Pragma' => 'no-cache',
+      'Cache-Control' => 'must-revalidate, post-check=0, pre-check=0',
+      'Expires' => '0',
+      'Accept-Ranges' => 'bytes',
+    ];
+
+    return new BinaryFileResponse($uri, 200, $headers);
+  }
+
+  /**
+   * Access callback for 'iCal' route.
+   */
+  public function access(GroupInterface $group) {
+    $access = $this->currentUser()->hasPermission('administer living spaces event');
+
+    return $access ? AccessResult::allowed() : AccessResult::forbidden();
```

Comment:
Would be better to have dedicated permission like "access ical export" both global and for groups so we have more flexibility who has access to exports.

Reply:
Added 'access ical event export' global and group permissions.